Simple Outputs/HSDP-1D.md

```diff
@@ -0,0 +1,152 @@
+
+```
+
+# HSDP-1D High Speed Data Printer
+
+| Item | Value | Comment |
+| :--- | :--- | :--- |
+| Vendor code | 0xf6976d00 | Chartronics, Inc. |
+| Device ID   | 0xcff2a11d | HSDP-1D High Speed Data Printer |
+| Device Type | 0xcff2     | Generic, nonstandard output device |
+| Version     | 0x1        | |
+
+## Description
+
+The HSDP line is all about printing the data you need as fast as possible.
+
+The HSDP line offers you the best way to store your data, in terms of cost
+efficency.
+
+Paper, power and data is all you need to run a HSDP. Its clever design combining
+the speed of the line matrix and the ink-free design of a spark printer will
+really cut the cost of your radio survey by orders of magnitude.
+
+The HSDP line is also very quiet - Chartronics guarantees less than 95dB of noise
+at full speed!
+
+The HSDP-1D is a Data version of the model 1. Equipped with cutting edge memory
+chips and buses, this printer will be able to log everything you throw at it.
+It is built to print large amounts of data in a smaller amount of time, though
+it lacks the typesetting possibilities of the -1T Typesetting variant.
+
+## Paper Format
+
+The paper used by the HSDP-1D is a continuous reel without natural "pages".
+
+Interrupt `2` can order the printer to cut the page after any line.
+
+Lines hold a maximum of 80 printed characters.
+
+## Speed
+
+The HSPD-1D prints 4 lines per second. (The content of the line doesn't matter,
+blank lines are not "skipped".)
+
+This works out to 240 lines per minute, or 4 pages per minute at 60 lines per
+page. (The HSPD-1D doesn't use pages, this is just for comparison with inferior
+printers from other manufacturers.)
+
+
+## Printing Modes
+
+The HSDP line is built around modes. The model 1D has three modes:
+
+| ID   | Mode Name | Content of one line       | Line Size |
+| :--- | :---      | :---                      | :---      |
+| 0    | Text Mode | 80 ASCII characters       | 80 words  |
+| 1    | Data Mode | 16 words                  | 16 words  |
+| 2    | Hex  Mode | 8 words                   |  8 words  |
+
+When ordered to print a line (see the interrupts below), the HSDP will read 80
+words (mode 0) or 16 words (mode 1) from the DCPU memory, and print a line
+containing that data.
+
+Mode 0 prints standard ASCII text. Control characters (including newlines,
+carriage returns and tabs) are ignored.
+
+Mode 1 prints 16 words formatted as hex with spaces between:
+
+```
+0000 1111 2222 3333 4444 5555 6666 7777  8888 9999 aaaa bbbb cccc dddd eeee ffff
+```
+
+Mode 2 prints 8 words formatted as hex, in a "hex dump" style:
+
+```
+80b0:  0000 1111 2222 3333  4444 5555 6666 7777    01234567
+````
+
+Where the rightmost block of text is the ASCII equivalents (where defined) or
+`.` if non-printing. (As with Mode 0 ASCII text, the upper 8 bits of each word
+are ignored for ASCII purposes.)
+
+
+## Interrupt Behavior
+
+- **0x0000: Set Mode** - Reads register `B` and sets the mode to  `B & 0x3`.
```

Comment:
Not sure about this. There may be eventually, in a different version. I could just say "sets the mode to B" and let it be undefined what happens if the mode is 78.